qa/rpc-tests/test_framework/util.py

```diff
@@ -355,9 +355,18 @@ def random_transaction(nodes, amount, min_fee, fee_increment, fee_variants):
 
     return (txid, signresult["hex"], fee)
 
-def assert_equal(thing1, thing2):
-    if thing1 != thing2:
-        raise AssertionError("%s != %s"%(str(thing1),str(thing2)))
+def assert_equal(expected, actual, message = ""):
+    if expected != actual:
+        if message:
+            message = "%s " % message 
```

Comment:
This looks like it will result in messages like `Should have received 1 note expected: <X> but was: <Y>`. Add a semicolon before the space.